src/spec.ts

```diff
@@ -50,8 +50,27 @@ export interface FacetSpec extends BaseSpec {
   spec: LayerSpec | UnitSpec;
 }
 
+export enum ResolveStrategy {
+  INDEPENDENT = 'independent' as any,
+  SHARED = 'shared' as any
+}
+
+export const INDEPENDENT = ResolveStrategy.INDEPENDENT;
+export const SHARED = ResolveStrategy.SHARED;
+
+export interface Resolve {
+  scale: ResolveStrategy;
+  guide?: ResolveStrategy;
```

Comment:
We don't use the term `guide` anywhere else in the syntax. 

I think we should use `axis` and `legend` here.